src/shogun/statistical_testing/kernelselection/internals/MaxCrossValidation.h

```diff
@@ -45,11 +45,12 @@ template <typename T> class SGVector;
 namespace internal
 {
 #ifndef DOXYGEN_SHOULD_SKIP_THIS
+template <typename PRNG>
 class MaxCrossValidation : public KernelSelection
```

Comment:
Making this class templated for PRNG is ugly, but I don't know what else to do (it's not CSGObject, and it implements the KernelSelection interface, in a method of which it uses PRNG)

Reply:
overall the internals of statistical_testing are uglier this way. they are not SGObjects, so can't use RandomMixin, and either the functions (if possible) if the class itself (as in this case) are templated to accept a reference of the PRNG